staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/validation/validation.go

```diff
@@ -720,8 +733,8 @@ func (v *specStandardValidatorV3) validate(schema *apiextensions.JSONSchemaProps
 	// WARNING: if anything new is allowed below, NewStructural must be adapted to support it.
 	//
 
-	if schema.Default != nil {
-		allErrs = append(allErrs, field.Forbidden(fldPath.Child("default"), "default is not supported"))
+	if schema.Default != nil && !v.allowDefaults {
+		allErrs = append(allErrs, field.Forbidden(fldPath.Child("default"), "must not be set"))
```

Comment:
suggested flipping the validation around at a higher level to entice structural schemas and pruning if they want to use default values, rather than just forbid the default values

see https://github.com/kubernetes/kubernetes/pull/77558/files#r287209541